dashboard/src/actions/repos.ts

```diff
@@ -287,3 +300,20 @@ export function checkChart(
     }
   };
 }
+
+export function fetchImagePullSecrets(
+  namespace: string,
+): ThunkAction<Promise<void>, IStoreState, null, AppReposAction> {
+  return async dispatch => {
+    try {
+      dispatch(requestImagePullSecrets(namespace));
+      const secrets = await Secret.list(namespace);
```

Comment:
IMO, ideally we'd not be pulling the full secrets across the net to the client (where MITM is a threat), but rather just getting the list of secret names (from a backend API or similar). It's a separate topic, but I'm not sure if Kubeapps could generally never see actual secrets client side (other than when they are created)? If it would be possible (ie. we only ever request the actual secret resource when explicitly asked to by the user to display in the UI), then we should create an issue and reference it here, IMO.

Reply:
well, this is not the only place in which we are fetching secrets. To be able to edit an AppRepository and modify the credentials used we are also fetching those. In any case, it's true that we can limit the scope of what the application receives.